### PR TITLE
[Epic #35961] Milestone 3 - Add Interactive Embedding CTA to Embed Modal

### PR DESCRIPTION
**Pull request by oisincoveney**
Epic #35961 

### Description

Adds the interactive embedding CTA to the new embed modal

### How to verify

As a Pro/EE user: clicking this banner should redirect you to the interactive embedding settings page
image

As an OSS user: clicking this banner should redirect you to the Embedded Analytics product page

image

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/public/components/widgets/SharingPane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+};
+
+export const InteractiveEmbeddingCTA = () => {
+  const isPaidPlan = useSelector(getIsPaidPlan);
+
+  const { showProBadge, description, linkText, linkTarget } =
+    getText(isPaidPlan);
+
+  const badge = (
+    // TODO: Check padding because design keeps using non-mantine-standard units
```
Comment (oisincoveney) [inline frontend/src/metabase/public/components/widgets/SharingPane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.tsx:39]: Just need to make sure on the spacing, maybe there's another component we can use that has these paddings. I'd use `Badge` but it doesn't seem to be ready yet

File: frontend/src/metabase/public/components/widgets/SharingPane/SharingPane.tsx [hunk 2]
```diff
@@ -69,56 +70,59 @@ function SharingPane({
...
-      <SharingPaneButton
-        header={t`Static embed`}
-        description={t`Securely embed this dashboard in your own application’s server code.`}
-        illustration={<StaticEmbedIcon />}
-        onClick={() => onChangeEmbedType("application")}
-      >
-        <SharingPaneActionButton
-          data-testid="sharing-pane-static-embed-button"
-          fullWidth
+    <Stack p="lg">
```
Comment (oisincoveney) [inline frontend/src/metabase/public/components/widgets/SharingPane/SharingPane.tsx:73]: The diff looks weird but I just nested everything into the `Stack` here and added the `InteractiveEmbeddingCTA` at the bottom

**Review by deniskaber — CHANGES_REQUESTED** [review]
Left inline comments. Most important one is that implementation differs from the design

File: frontend/src/metabase/public/components/widgets/SharingPane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.unit.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        "token-status": createMockTokenStatus({ valid: isPaidPlan }),
+      }),
+    }),
+  });
+
+  return {
+    history: checkNotNull(history),
+  };
+};
+describe("InteractiveEmbeddingCTA", () => {
```
[the remark this answers is not in the document]
Comment (oisincoveney) [reply frontend/src/metabase/public/components/widgets/SharingPane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.unit.spec.tsx:18]: Yep, good call. I'll add those to the existing tests in the embed modal

File: frontend/src/metabase/public/components/widgets/SharingPane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.styled.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+export const ProBadge = styled(Group)`
+  border-radius: ${({ theme }) => theme.radius.xs};
+`;
+
+export const CTAContainer = styled(Paper)<PaperProps>``;
+
+export const ClickIcon = styled(Icon)`
+  ${CTAContainer}:hover & {
+    color: ${({ theme }) => theme.colors.brand[1]};
```
[the remark this answers is not in the document]
Comment (oisincoveney) [reply frontend/src/metabase/public/components/widgets/SharingPane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.styled.tsx:14]: Yeah, it's a weird one but the `theme` refers to the Mantine color scheme that we're trying to migrate to, so we basically just have to deal with it unfortunately

File: frontend/src/metabase/public/components/widgets/SharingPane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        span
+        tt="uppercase"
+        lts="0.7"
+        size="sm"
+        fw={700}
+      >{t`Pro`}</Text>
+    </ProBadge>
+  );
+
+  return (
```
[the remark this answers is not in the document]
Comment (oisincoveney) [reply frontend/src/metabase/public/components/widgets/SharingPane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.tsx:53]: Wow, I didn't realize how far I strayed away from the designs. Thanks for the callout, I'll make sure it matches.

File: frontend/src/metabase/public/components/widgets/SharingPane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      linkText: t`Set it up`,
+      linkTarget: "/admin/settings/embedding-in-other-applications/full-app",
+    };
+  }
+
+  return {
+    showProBadge: true,
+    description: t`Give your customers the full power of Metabase in your own app, with SSO, advanced permissions, customization, and more.`,
+    linkText: t`Learn more`,
+    linkTarget: "https://www.metabase.com/product/embedded-analytics",
```
[the remark this answers is not in the document]
Comment (oisincoveney) [reply frontend/src/metabase/public/components/widgets/SharingPane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.tsx:28]: I don't think we can actually use `getDocsUrl` here - the `/product/embedded-analytics` page isn't a doc URL, but rather a marketing page

File: frontend/src/metabase/public/components/widgets/SharingPane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (oisincoveney) [reply frontend/src/metabase/public/components/widgets/SharingPane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.tsx:54]: @deniskaber let me know if this matches your expectations a little better 😄

**Review by albertoperdomo — CHANGES_REQUESTED** [review]
I think it looks very close, but not quite like the design. What I notice is some issue with vertical alignment of icon and text.

Screenshots below

Design: 
image

Implementation:
image

**Review by albertoperdomo — APPROVED** [review]

**Review by deniskaber — APPROVED** [review]
Approve. But I also found one more issue while testing this - 
Screenshot 2023-12-20 at 20 28 09

Sharing / Embedding popover has scrollbars

**Review by vbenedetti — APPROVED** [review]

**Pull request merged**